packages/core/test/integration/query-interface/add-show-remove-index.test.ts

```diff
@@ -0,0 +1,304 @@
+        }
+      });
+
+      it('should add, show and remove unique index with include columns', async () => {
+        await queryInterface.addIndex('users', { name: 'user_username', fields: ['username'], include: ['first_name', 'last_name'], unique: true });
+
+        const indexes = await sequelize.queryInterface.showIndexes('users');
+        const indexCheck = indexes.find(index => index.name === 'user_username');
+
+        expect(indexCheck).to.not.be.undefined;
+        assert(indexCheck);
+        expect(indexCheck.tableName).to.equal('users');
+        expect(indexCheck.name).to.equal('user_username');
+        expect(indexCheck.unique).to.equal(true);
+        expect(indexCheck.fields).to.have.length(1);
+        expect(indexCheck.fields[0].name).to.equal('username');
+        expect(indexCheck.includes).to.have.length(2);
+        expect(indexCheck.includes).to.include('first_name');
+        expect(indexCheck.includes).to.include('last_name');
+
+        await queryInterface.removeIndex('users', 'user_username');
+        const indexesAfterRemoval = await sequelize.queryInterface.showIndexes('users');
+        const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'user_username');
+
+        expect(indexCheckAfterRemoval).to.be.undefined;
+      });
+
+      it('throws an error with duplicate column names in included columns', async () => {
+        try {
+          await queryInterface.addIndex('users', { name: 'user_username', fields: ['username'], include: ['username', 'first_name', 'last_name'], unique: true });
+          if (dialect === 'postgres') {
+            const indexes = await sequelize.queryInterface.showIndexes('users');
+            const indexCheck = indexes.find(index => index.name === 'user_username');
+
+            expect(indexCheck).to.not.be.undefined;
+            assert(indexCheck);
+            expect(indexCheck.tableName).to.equal('users');
+            expect(indexCheck.name).to.equal('user_username');
+            expect(indexCheck.unique).to.equal(true);
+            expect(indexCheck.fields).to.have.length(1);
+            expect(indexCheck.fields[0].name).to.equal('username');
+            expect(indexCheck.includes).to.have.length(3);
+            expect(indexCheck.includes).to.include('username');
+            expect(indexCheck.includes).to.include('first_name');
+            expect(indexCheck.includes).to.include('last_name');
+
+            await queryInterface.removeIndex('users', 'user_username');
+            const indexesAfterRemoval = await sequelize.queryInterface.showIndexes('users');
+            const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'user_username');
+
+            expect(indexCheckAfterRemoval).to.be.undefined;
+          } else {
+            expect.fail('This should have failed');
+          }
+        } catch (error) {
```

Comment:
this just tests database limits instead of the ORM, I don't think this test is relevant for us

Reply:
Updated now.